File: app/src/main/java/com/mxt/anitrend/model/entity/anilist/MediaListCollection.java
package com.mxt.anitrend.model.entity.anilist;

import java.util.List;
import android.os.Parcel;
import com.mxt.anitrend.model.entity.base.MediaListCollectionBase;

public class MediaListCollection extends MediaListCollectionBase {

    private List<MediaList> entries;

    protected MediaListCollection(Parcel in) {
        super(in);
        entries = in.createTypedArrayList(MediaList.CREATOR);
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
        dest.writeTypedList(entries);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<MediaListCollection> CREATOR = new Creator<MediaListCollection>() {
        @Override
        public MediaListCollection createFromParcel(Parcel in) {
            return new MediaListCollection(in);
        }

        @Override
        public MediaListCollection[] newArray(int size) {
            return new MediaListCollection[size];
        }
    };

    public List<MediaList> getEntries() {
        return entries;
    }
}

File: app/src/main/java/com/mxt/anitrend/model/entity/anilist/MediaListCollection.kt
package com.mxt.anitrend.model.entity.anilist

import android.os.Parcel
import android.os.Parcelable
import com.mxt.anitrend.model.entity.base.MediaListCollectionBase

class MediaListCollection : MediaListCollectionBase {
    var entries: List<MediaList>? = null
        private set

    protected constructor(parcel: Parcel) : super(parcel) {
        entries = parcel.createTypedArrayList(MediaList.CREATOR)
    }

    override fun writeToParcel(
        dest: Parcel,
        flags: Int,
    ) {
        super.writeToParcel(dest, flags)
        dest.writeTypedList(entries)
    }

    override fun describeContents(): Int = 0

    companion object CREATOR : Parcelable.Creator<MediaListCollection> {
        override fun createFromParcel(parcel: Parcel): MediaListCollection = MediaListCollection(parcel)

        override fun newArray(size: Int): Array<MediaListCollection?> = arrayOfNulls(size)
    }
}
